src/app/app.component.spec.ts

```diff
@@ -1,17 +1,52 @@
-import { TestBed } from '@angular/core/testing';
+import {ComponentFixture, TestBed} from '@angular/core/testing';
 import { RouterTestingModule } from '@angular/router/testing';
 import { AppComponent } from './app.component';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
+import {WeatherServiceService} from './weather-service.service';
+import {of} from 'rxjs';
+import {WeatherDataComponent} from './weather-data/weather-data.component';
 
 describe('AppComponent', () => {
+
+  let component: AppComponent;
+  let fixture: ComponentFixture<AppComponent>;
+  let service: WeatherServiceService;
+  let httpTestingController: HttpTestingController;
+  let mockData = 1;
+  let mockResult:any = [];
+
   beforeEach(async () => {
     await TestBed.configureTestingModule({
       imports: [
+        HttpClientTestingModule,
         RouterTestingModule
       ],
       declarations: [
-        AppComponent
+        AppComponent,
+        WeatherDataComponent
       ],
+      providers:[
+        WeatherServiceService
+      ]
     }).compileComponents();
   });
 
+  beforeEach(() => {
+    service = TestBed.inject(WeatherServiceService);
+    httpTestingController = TestBed.inject(HttpTestingController);
+    fixture = TestBed.createComponent(AppComponent);
+    component = fixture.componentInstance;
+  })
+
+  it('getData', async () => {
+    service.getLocationData = jest.fn().mockImplementation( () => {
+      component.weatherData.push(mockData);
```

Comment:
I don't think this line should be there, as the service it's self isn't doing this